### ISSUE TITLE
xow stops working after suspending and waking up the computer

### ISSUE THREAD

**Issue by webodan** [medusalix/xow#39]
Hi,

Basically title is what's happening. I've noticed that after waking up the computer and trying to sync the controller, xow would not work, so I rebooted the service using `systemctl` and it's working fine again. This might not be appropriate for a retropie style build like the one I'm trying to make, so it would be fine if the program was able to be aware that it resumed from sleep and kept working. 

Here's the output of the program as the computer goes to sleep and wakes up:

> dan@retromint:~$ xow
> 2020-02-26 14:35:23 INFO  - xow  ©Severin v. W.
> 2020-02-26 14:35:23 INFO  - Dongle plugged in
> 2020-02-26 14:35:23 INFO  - Wireless address: 62:45:b5:0d:5b:71
> 2020-02-26 14:35:23 INFO  - Dongle initialized
> -- PC GOES TO SLEEP AND I WAKE IT UP AFTER A WHILE --
> 2020-02-26 14:36:28 ERROR - Error in bulk read: 5
> 2020-02-26 14:36:28 ERROR - Error in bulk read: 5
> 

I'm using the reliability branch btw.
Also a big thank you for the author of this software as I'm finally able to use my xbox controller wirelessly on Linux! really hope audio is implemented sometime too <3

**Comment by medusalix** [comment]
I've done some testing and it seems like `libusb` is returning `LIBUSB_TRANSFER_NO_DEVICE` after waking up the computer. This usually only happens when you physically remove the dongle. I'll probably have to make some adjustments to xow to handle this case.

**Comment by medusalix** [comment]
@webodan Please try out the [`suspend`](https://github.com/medusalix/xow/tree/suspend) branch (make sure that the updated service definition gets loaded). While I couldn't find a way for xow to recover after a suspend, I made sure that the service restarts when the system wakes up.

**Comment by medusalix** [comment]
@NoXPhasma and @andreashuetter, could you try this out for me? I made some significant changes to the USB backend (making it simpler) which will definitely need testing. As an inevitable result of fixing the suspend issue, xow does now terminate when you wake from sleep or unplug the dongle and is restarted automatically by `systemd`.

**Comment by NoXPhasma** [comment]
The service stops when the dongle gets removed, but does not restart when I plug in the dongle again.

```
Mär 15 04:30:31 NoXP-PC systemd[1]: Started Xbox One Wireless Dongle Driver.
Mär 15 04:30:31 NoXP-PC xow[533968]: 2020-03-15 04:30:31 INFO  - xow v0.3-27-g28cbc4d ©Severin v. W.
Mär 15 04:30:31 NoXP-PC xow[533968]: 2020-03-15 04:30:31 INFO  - Dongle plugged in
Mär 15 04:30:31 NoXP-PC xow[533968]: 2020-03-15 04:30:31 INFO  - Wireless address: 62:45:b4:fe:81:16
Mär 15 04:30:31 NoXP-PC xow[533968]: 2020-03-15 04:30:31 INFO  - Dongle initialized
Mär 15 04:30:35 NoXP-PC xow[533968]: 2020-03-15 04:30:35 ERROR - Error in bulk read: LIBUSB_ERROR_NO_DEVICE
Mär 15 04:30:35 NoXP-PC xow[533968]: 2020-03-15 04:30:35 ERROR - Error in bulk read: LIBUSB_ERROR_NO_DEVICE
Mär 15 04:30:35 NoXP-PC systemd[1]: xow.service: Succeeded.
```

**Comment by medusalix** [comment]
Make sure that you run `systemctl enable xow` and `systemctl restart xow` after installing the branch to reload the service config.

**Comment by NoXPhasma** [comment]
Yes, I've done that. It just doesn't restart itself after I remove the dongle and reconnect it.

**Comment by NoXPhasma** [comment]
Okay I found the issue, `sudo make install` won't install the new service file which includes `Restart=always`. Once I update the service file manually it works.

**Comment by medusalix** [comment]
It should override them. When I checkout `master`, do `install` and then checkout `suspend` followed by `install` it replaces the contents of the service.

**Comment by NoXPhasma** [comment]
I expected the same here as well, but for some reason it didn't. I even tried to install it two times to get sure.

**Comment by medusalix** [comment]
Oh well, I'm just going to add a notice to run `sudo make uninstall` to the README to make sure everyone's covered.

**Comment by medusalix** [comment]
Merged into `master` (196371dc40683e52af572347a94bd8193fa2a28a).
